pkg/workflows/wasm/host/module.go

```diff
@@ -326,25 +454,49 @@ func (m *Module) Run(ctx context.Context, request *wasmpb.Request) (*wasmpb.Resp
 		return nil, fmt.Errorf("invalid request: can't be empty")
 	}
 
+	if !m.isLegacyDAG {
+		return m.wrapRunForNoDAG(ctx, request)
+	}
+
+	return runWasm[*wasmdagpb.Request, *wasmdagpb.Response](m, ctx, request, m.requestStore)
+}
+
+type idMessage interface {
+	GetId() string
+	proto.Message
+}
+
+func runWasm[I idMessage, O proto.Message](
```

Comment:
what is runWasm for?

Reply:
I refactored the Run method that's used for legacy into a helper method so that the DAG and no DAG share common code.